PostHog/PostHogFileBackedQueue.swift

```diff
@@ -100,10 +100,14 @@ class PostHogFileBackedQueue {
 
     private func deleteFiles(_ count: Int) {
         for _ in 0 ..< count {
-            if items.isEmpty { return }
-            let removed = items.remove(at: 0) // We always remove from the top of the queue
-
-            deleteSafely(queue.appendingPathComponent(removed))
+            if let removed: String = _items.mutate({ items in
+                if items.isEmpty {
+                    return nil
+                }
+                return items.remove(at: 0) // We always remove from the top of the queue
```

Comment:
How did you reproduce this issue since the whole block is locked with the `isFlushingLock` and `isFlushing` flag? Can you pass the specific steps to reproduce the issue so we can check if the very same issue happens in more places as well?
I don't see how the `remove(at: 0)` would fail if the internal `items` list isn't modified anywhere else outside of this lock.

Reply:
Checking the thread is easy (`Thread.isMainThread`), but independently of the thread, the add method should be run within the `dispatchQueue.async` block and then it's ok to lock with `isFlushingLock` and await its availability.
Again, another PR, fixing the crash here is more important now, but a follow-up with the improvement above would be cool.
https://github.com/PostHog/posthog-ios/blob/1bb1709031254f1e243f5a0de57fc9e969ec8c65/PostHog/PostHogQueue.swift#L212 is an edge case so I bet rarely happens

Reply:
fwiw I only got one crash report with this call stack.